snabbdom/src/main/scala/snabbdom/VNodeData.scala

```diff
@@ -38,90 +38,23 @@
 
 package snabbdom
 
-class VNodeData(
-    var props: Option[Map[String, PropValue]],
-    var attrs: Option[Map[String, AttrValue]],
-    var classes: Option[Map[String, ClassValue]],
-    var style: Option[Map[String, StyleValue]],
-    var dataset: Option[Map[String, String]],
-    var on: Option[Map[String, EventHandler]],
-    var hook: Option[Hooks],
-    var key: Option[String],
-    var ns: Option[String], // for SVG
-    var fn: Option[Seq[Any] => VNode], // for thunks
-    var args: Option[Seq[Any]], // for thunks
-    var is: Option[String]
+case class VNodeData(
+    val props: Map[String, PropValue] = Map.empty,
+    val attrs: Map[String, AttrValue] = Map.empty,
+    val classes: Map[String, ClassValue] = Map.empty,
+    val style: Map[String, StyleValue] = Map.empty,
+    val dataset: Map[String, String] = Map.empty,
+    val on: Map[String, EventHandler] = Map.empty,
+    val hook: Option[Hooks] = None,
+    val key: Option[String] = None,
+    val ns: Option[String] = None, // for SVG
+    val fn: Option[Seq[Any] => VNode] = None, // for thunks
+    val args: Option[Seq[Any]] = None, // for thunks
+    val is: Option[String] = None
```

Comment:
Also, `case class`es are sadly notoriously difficult to maintain bincompat for (e.g., if we want to add a new field at some point). But I'll leave that problem for the future 😅 

```suggestion
case class VNodeData(
    props: Map[String, PropValue] = Map.empty,
    attrs: Map[String, AttrValue] = Map.empty,
    classes: Map[String, ClassValue] = Map.empty,
    style: Map[String, StyleValue] = Map.empty,
    dataset: Map[String, String] = Map.empty,
    on: Map[String, EventHandler] = Map.empty,
    hook: Option[Hooks] = None,
    key: Option[String] = None,
    ns: Option[String] = None, // for SVG
    fn: Option[Seq[Any] => VNode] = None, // for thunks
    args: Option[Seq[Any]] = None, // for thunks
    is: Option[String] = None
```